frontend/ui/elements/screensaver_menu.lua

```diff
@@ -64,6 +64,16 @@ return {
                             Screensaver:setStretchLimit(touchmenu_instance)
                         end,
                     },
+                    {
+                        text = _("Rotate cover to fit screen"),
+                        checked_func = function()
+                            return G_reader_settings:isTrue("screensaver_autorotate_images")
+                        end,
+                        callback = function(touchmenu_instance)
+                            G_reader_settings:flipNilOrFalse("screensaver_autorotate_images")
+                            touchmenu_instance:updateItems()
+                        end,
+                    }
```

Comment:
this should be nested under `Custom images` not `Border fill`.

Reply:
I suppose a case could also be made in favor of reusing `imageviewer_rotate_auto_for_best_fit` instead of duplicating a ScreenSaver-specific variant?

Reply:
Dunno if that setting should be reused - but the new one may keep the same name suffix:
`screensaver_rotate_auto_for_best_fit`

Reply:
@Commodore64user It would be confusing to have stretch and rotate in different menus. Maybe move both to `Custom images`?
@NiLuJe I don't think it should be tied to the image viewer. A user might want their screensavers to look pretty and fill the screen, while preferring images to stay upright in the reader.
@poire-z Sounds good

Reply:
Fair enough, I'm not the target audience for either use-cases, so that's perfectly fine by me ;).

Reply:
> @Commodore64user It would be confusing to have stretch and rotate in different menus. Maybe move both to `Custom images`?

On the other hand, this *only* applies to custom images, while the border fill stuff applies to *all* images ;).

Reply:
Rotation applies to both covers and custom images

Reply:
Oh, right, my bad, hadn't re-read the code ;).

Then, err, meh, no strong feelings about either ;p.

Reply:
It's a big chunky if statement, a comment is in order. Commodore64user, what do you think?

Reply:
> Maybe move both to Custom images

no, to that one.

Are covers in the wrong orientation…? In any case border fill has things related to adjusting (or reducing them) borders, while rotation is an entirely different thing… perhaps you could rename it, to border fill and rotation but i am not convinced (thinking about it)

Reply:
The more i think about this, the more i realise it belongs in “custom images” (sorry if you need to move from one place to another to adjust things). I am all for people having cats, pandas or cows as their sleep screen image but let’s keep things where they belong. Otherwise we’ll just end up with the mess things become when one starts leaving them in the “everyday” table.

Reply:
The rotation is meant to reduce borders, at least that was the use case I had in mind (it never rotates if the borders would become wider). "Border fill" already read awkwardly to me - it relies on "fill" being both a noun (black fill) and a verb (fill the screen). How about "Border fill and fit"? That matches the language used in the options.

Reply:
Does it relate to everything or only to a custom image?

Reply:
> The rotation is meant to reduce borders, at 

This is just you trying to force the wrong piece in the wrong spot.

> Border fill" already read awkwardly to me

Border and background fill are legitimate terms used in image retouching and editing.

> Does it relate to everything or only to a custom image?

This is what i have been trying to get at...
Are you solving #12308 as well? If yes then we should rename the parent, if not, it goes with "custom images"

Reply:
> This is just you trying to force the wrong piece in the wrong spot. Not buying it
I made this feature specifically to reduce pillarboxing, i.e., borders, and as you said, border fill is for adjusting or reducing borders. I don't know what else you think this feature is for.

> Border and background fill are legitimate terms used in image retouching and editing.
They refer to stretching an image to fill borders?

> Does it relate to everything or only to a custom image?
> Rotation applies to both covers and custom images.

This is what i have been trying to get at...
Are you solving #12308 as well? If yes then we should rename the parent, if not, it goes with "custom images"
That issue seems to be specific to devices with hardware rotation (not sure) but inverted portrait normally works fine without rotation.

Reply:
Border fill is just to paint the border, or fill the borders with black or white. Stretch image just seems to be in the same direction. Filling the borders with a stretched out cover.

> I made this feature specifically to reduce pillarboxing,

could you show screenshots with this on and off… 

> That issue seems to be specific to devices with hardware rotation (not sure) but inverted portrait normally works fine without rotation

that is what i was afraid of, the only reason i can see for rotating a book cover is this as far as i can see.

Reply:
I don't think https://github.com/koreader/koreader/issues/12308 is really the same thing as "I want it to fill up as much space as possible regardless of rotation"

Reply:
Another option could be to use the same enabled function as custom images https://github.com/koreader/koreader/blob/91c15d52778708b580e0bd0bcc6e364051e98cbc/frontend/ui/elements/screensaver_menu.lua#L82-L85 because i can't still fathom why anyone would want to use this with a legitimate book cover.

Reply:
I'll upload screenshots later today.

Off the top of my head, children's picture books and PDF slideshows have horizontal covers. The option also rotates vertical covers on a horizontal device, though I can't confirm right now if KOReader considers any devices to have a horizontal resolution (tablets?)

Anyway, why limit it to images when it works for both? Vertical covers aren't affected whether it's on or off.

Reply:
I'm for "Border fill".

Reply:
> "Border fill and fit"?

Sounds good to me!

> Does it relate to everything or only to a custom image?

Everything, which is why moving this in "Custom images" would be problematic.

Reply:
> though I can't confirm right now if KOReader considers any devices to have a horizontal resolution (tablets?)

We don't discriminate ;p.

Whatever the native layout is happens to be "Portrait" ;p. (i.e., on a TV, UR would look landscape-y).

Reply:
Rest assured that very very few things are left using getScreenMode in the codebase ;).

Reply:
> Off the top of my head, children's picture books and PDF slideshows have horizontal covers. 

not sure there are many 3 year olds using KOReader though. as for slideshows, not books.

> The option also rotates vertical covers on a horizontal device,

which again, I don't personally understand why anyone would want this done. But that just might be me.

> Anyway, why limit it to images when it works for both? 

because it might be possible, but whether it should be done is a completely different issue, then again we (or I) can only be sure with examples.

> Vertical covers aren't affected whether it's on or off.

but this here just _might_ be how you snatch victory from the jaws of defeat.

Reply:
> because it might be possible, but whether it should be done is a completely different issue, then again we (or I) can only be sure with examples.

PDFs might be a legitimate use-cases where te cover (or first page when lacking) one is genuinely in a different orientation than the device.

Say, for example, a landscape page on a Portrait device ;).

Reply:
> PDFs might be a legitimate use-cases where the cover (or first page when lacking one) is genuinely in a different orientation than the device.

that might seal the deal then. I _might_ have to stand corrected then.

Reply:
Alright then, I'll mark this comment thread as resolved for now and change the menu name.

Reply:
not yet, still waiting for those screenshots.